python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -508,7 +512,15 @@ def updateBlendRecords(blendData, catalog, modelPsf, observedPsf, redistributeIm
             # Set the flux at the center of the model
             peak = heavy.peaks[0]
             img = heavyFootprintToImage(heavy, fill=0.0)
-            sourceRecord.set("deblend_peak_instFlux", img.image[Point2I(peak["i_x"], peak["i_y"])])
+            try:
+                sourceRecord.set("deblend_peak_instFlux", img.image[Point2I(peak["i_x"], peak["i_y"])])
+            except Exception:
+                srcId = sourceRecord.getId()
+                x = peak["i_x"]
+                y = peak["i_y"]
+                logger.warn(f"Source {srcId} at {x},{y} could not set the peak flux with error:")
+                traceback.print_exc()
```

Comment:
This should not be called because it won't go to the logging system and we can't tell where it will turn up. Instead you should use `logger.exception()` and that would print your log message above and also include the exception traceback.